### PR TITLE
PSCC paper: Updated xfmr_exc structure

### LINKED ISSUE #1: PSCC paper: Gradient of the excitation current
The gradient, also the Hessian but not of interest given JuMP limitations, of an interpolation is freely available which means that the autodiff setting could be avoided, possibly speeding up computation.

Issue: to get the gradient at a node `x` of an interpolation `int`, the following is required: `grad = gradient(int,x...)`, which introduces an extra argument in the function call, making it impossible to register as is (status 29/07/21).

Possible fix: add function to the power models data file rather than the interpolation, where `int` is already parsed as an argument.

### LINKED ISSUE #16: PSCC Paper: generalize BH curve input
Checklist:
- [ ] Correct angle fault in the flux definition
- [ ] Redefine the input dictionary, allowing for multiple inputs
- [ ] Document the process

### PR DESCRIPTION
**Pull request by timmyfaraday**
notes:
- magnetic flux density function needs to be reviewed for consistent units
- is e(t) = sin(wt + theta)??
- TODO: excitation bounds for excitation voltage of non-considered harmonics should be taken from the fr-bus
- TODO: automated tests

### CODE REVIEW

**Comment by frederikgeth** [comment]
can we merge PR #11 first ?

File: examples/excitation_input.jl [hunk 1]
```diff
@@ -0,0 +1,8 @@
+## BH-CURVE EXAMPLE
+# pkgs
+using Dierckx
+
+# Thyssenkrupp - PowerCore H 100-23
+B = [0.000, 0.144, 0.200, 0.260, 0.328, 0.400, 0.504, 0.600, 0.695, 1.528, 1.716, 1.776, 1.816, 1.828, 1.832, 1.845, 1.856, 1.860]
```
Comment (frederikgeth) [inline examples/excitation_input.jl:6]: is there a reference / citation for this information?

**Comment by timmyfaraday** [comment]
> can we merge PR #11 first ?

done

**Comment by frederikgeth** [comment]
ready to merge IMO, just gotta fix the conflict.

**Review by frederikgeth — APPROVED** [review]
great adaptation. also very excited about the docstrings, this is a good motivator to set up actual docs.

File: examples/excitation_input.jl (hunk 1 above)
Comment (timmyfaraday) [reply examples/excitation_input.jl:6]: The BH-curves I send you a while ago, however, I can't currently find them on the Thyssenkrupp website. I'll make an inquiry!

Comment (timmyfaraday) [reply examples/excitation_input.jl:6]: Example:

https://www.thyssenkrupp-steel.com/media/content_1/publikationen/produktinformationen/powercore_no/thyssenkrupp_powercore_020-130y320_product_information_steel_en.pdf

**Pull request merged**